lib/libc/minimal/include/time.h

```diff
@@ -4,7 +4,41 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_TIME_H_
+#define ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_TIME_H_
+
+#include <stdint.h>
 
 /* Dummy time.h to fulfill the requirements of certain libraries
- * i.e. mbedTLS
+ * i.e. mbedTLS and Zephyr civil time APIs
  */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct tm {
+	int tm_sec;
+	int tm_min;
+	int tm_hour;
+	int tm_mday;
+	int tm_mon;
+	int tm_year;
+	int tm_wday;
+	int tm_yday;
+	int tm_isdst;
+};
+
+typedef int64_t time_t;
+typedef int32_t suseconds_t;
+
+struct timespec {
```

Comment:
This conflicts with https://github.com/zephyrproject-rtos/zephyr/pull/16626 , which takes care about minlibc vs newlib compatibility.